jinahub/encoders/text/CLIPTextEncoder/clip_text.py

```diff
@@ -1,69 +1,128 @@
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        if default_traversal_paths is not None:
+            self.default_traversal_paths = default_traversal_paths
+        else:
+            self.default_traversal_paths = ['r']
+        self.default_batch_size = default_batch_size
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.base_tokenizer_model = (
+            base_tokenizer_model or pretrained_model_name_or_path
+        )
+        self.max_length = max_length
+        self.logger = JinaLogger(self.__class__.__name__)
+
+        if device not in ['cpu', 'cuda']:
+            self.logger.error('Torch device not supported. Must be cpu or cuda!')
+            raise RuntimeError('Torch device not supported. Must be cpu or cuda!')
+
+        if device == 'cuda' and not torch.cuda.is_available():
+            self.logger.warning(
+                'You tried to use GPU but torch did not detect your'
+                'GPU correctly. Defaulting to CPU. Check your CUDA installation!'
+            )
+            device = 'cpu'
+
+        if device == 'cpu' and num_threads:
+            cpu_num = os.cpu_count()
+            if num_threads > cpu_num:
+                self.logger.warning(
+                    f'You tried to use {num_threads} threads > {cpu_num} CPUs'
+                )
+            else:
+                torch.set_num_threads(num_threads)
+
+        self.device = device
+        self.embedding_fn_name = embedding_fn_name
+        self.tokenizer = CLIPTokenizer.from_pretrained(self.base_tokenizer_model)
+        self.model = CLIPModel.from_pretrained(self.pretrained_model_name_or_path)
+        self.model.to(torch.device(device))
+
+    @requests
+    def encode(self, docs: Optional[DocumentArray], parameters: Dict, **kwargs):
+        """
+        Encode text data into a ndarray of `D` as dimension, and fill
+        the embedding attribute of the docs.
+
+        :param docs: DocumentArray containing text
+        :param parameters: dictionary to define the `traversal_paths` and the `batch_size`. For example,
+               `parameters={'traversal_paths': ['r'], 'batch_size': 10}`.
+        :param kwargs: Additional key value arguments.
+        """
+        for document_batch in get_docs_batch_generator(
+            docs,
+            traversal_path=parameters.get(
+                'traversal_paths', self.default_traversal_paths
+            ),
+            batch_size=parameters.get('batch_size', self.default_batch_size),
+            needs_attr='text',
+        ):
+            text_batch = document_batch.get_attributes('text')
+
+            with torch.no_grad():
+                input_tokens = self._generate_input_tokens(text_batch)
+                embedding_batch = getattr(self.model, self.embedding_fn_name)(
+                    **input_tokens
+                )
+                if isinstance(embedding_batch, torch.Tensor):

```

Comment:
I think this is not needed - we have in in transformers encoder as there can be different models with different outputs, but now we have only one model, we should know what it returns